templates/swagger/v1_json.tmpl

```diff
@@ -18921,6 +18961,22 @@
       },
       "x-go-package": "code.gitea.io/gitea/modules/structs"
     },
+    "RenameUserOption": {
+      "description": "CreateKeyOption options when creating a key",
+      "type": "object",
+      "required": [
+        "new_username"
+      ],
+      "properties": {
+        "new_username": {
+          "description": "New username",
```

Comment:
This description doesn't add any information beyond what's already known from the property name. How about something like "New username for this user. This name cannot be in use yet by any other user."

IMO it should also include information about whether it is a valid operation to 'rename' to the same username, say a user `sybren` is renamed with `new_username: "sybren"`. Would that be an acceptable operation (but a no-op) or would it cause an error?

Reply:
Thanks. I've updated the description. I'm still on the fence of if it should return an error or not. I'm going to test the UI and align it with whatever it does (show error or not)

Reply:
The UI doesn't return an error as the form checks which fields have changed then updates accordingly, so that if the username remains the same then the rename code isn't touched. Therefore I will update this to treat it as a noop and not return any error.